clang/labs/lab1/kruglov_alexey/RenameID.cpp

```diff
@@ -0,0 +1,166 @@
+#include "clang/AST/ASTConsumer.h"
+#include "clang/AST/RecursiveASTVisitor.h"
+#include "clang/Frontend/CompilerInstance.h"
+#include "clang/Frontend/FrontendPluginRegistry.h"
+#include "clang/Rewrite/Core/Rewriter.h"
+#include "llvm/ADT/Sequence.h"
+
+class RenameVisitor : public clang::RecursiveASTVisitor<RenameVisitor> {
+private:
+  clang::Rewriter Rewriter;
+  std::string OldName;
+  std::string NewName;
+
+public:
+  explicit RenameVisitor(clang::Rewriter Rewriter, std::string OldName,
+                         std::string NewName)
+      : Rewriter(Rewriter), OldName(OldName), NewName(NewName){};
+
+  bool VisitFunctionDecl(clang::FunctionDecl *Func) {
+    std::string Name = Func->getNameAsString();
+
+    if (Name == OldName) {
+      Rewriter.ReplaceText(Func->getNameInfo().getSourceRange(), NewName);
+    }
+    if (Func->getReturnType().getAsString() == OldName ||
+        Func->getReturnType().getAsString() == OldName + " *" ||
+        Func->getReturnType().getAsString() == OldName + " &") {
```

Comment:
This code is repeated several times across this class. Please, extract it to the separate function